comms/dht/src/connectivity/mod.rs

```diff
@@ -336,6 +339,27 @@ impl DhtConnectivity {
         Ok(())
     }
 
+    async fn request_many_dials_with_initial_delay<I: IntoIterator<Item = NodeId>>(
+        &mut self,
+        peers: I,
+    ) -> Result<(), DhtConnectivityError> {
+        self.handle_initial_delay().await;
+        self.connectivity.request_many_dials(peers).await?;
+        Ok(())
+    }
+
+    async fn handle_initial_delay(&mut self) {
+        if !self.first_round_completed {
+            self.first_round_completed = true;
```

Comment:
Something about this feels like a code smell. The bool causes a delay in the first round. Can't we somehow apply that to the first round only, without requesting a function `request_many_dials_with_initial_delay` that implies a delay although we never actually expect the delay to occur again.

Reply:
Yeah I agree, this delay should not be in this layer, the caller should call this when they mean to, e.g after a delay